tests/test_unit/test_napari_plugin/test_poses_loader_widget.py

```diff
@@ -0,0 +1,170 @@
+    assert source_software_combo.count() == len(SUPPORTED_POSES_FILES)
+
+    # Test that the default fps is 30
+    fps_spinbox = poses_loader_widget.findChildren(QSpinBox)[0]
+    assert fps_spinbox.value() == 30
+
+    # Make sure that the line edit for file path is empty
+    file_path_edit = poses_loader_widget.findChildren(QLineEdit)[-1]
+    assert file_path_edit.text() == ""
+
+    # Make sure that the first button is a "Browse" button
+    browse_button = poses_loader_widget.findChildren(QPushButton)[0]
+    assert browse_button.text() == "Browse"
+
+    # Make sure that the last row is a "Load" button
+    load_button = poses_loader_widget.findChildren(QPushButton)[-1]
+    assert load_button.text() == "Load"
+
+    # Make sure that layer tooltips are enabled
+    assert get_settings().appearance.layer_tooltip_visibility is True
+
+
+# --------test connection between widget buttons and methods------------------#
+def test_browse_button_calls_on_browse_clicked(
+    make_napari_viewer_proxy, mocker
+):
+    """Test that clicking the 'Browse' button calls the right function."""
+    mock_method = mocker.patch(
+        "movement.napari._loader_widgets.PosesLoader._on_browse_clicked"
+    )
+    poses_loader_widget = PosesLoader(make_napari_viewer_proxy)
+    browse_button = poses_loader_widget.findChildren(QPushButton)[0]
+    browse_button.click()
+    mock_method.assert_called_once()
+
+
+def test_load_button_calls_on_load_clicked(make_napari_viewer_proxy, mocker):
+    """Test that clicking the 'Load' button calls the right function."""
+    mock_method = mocker.patch(
+        "movement.napari._loader_widgets.PosesLoader._on_load_clicked"
+    )
+    poses_loader_widget = PosesLoader(make_napari_viewer_proxy)
+    load_button = poses_loader_widget.findChildren(QPushButton)[-1]
+    load_button.click()
+    mock_method.assert_called_once()
+
+
+# ------------------- tests for widget methods--------------------------------#
+# In these tests we check if calling a widget method has the expected effects
+
+
+@pytest.mark.parametrize(
+    "file_path",
+    [
+        # valid file path
+        str(DATA_PATHS.get("DLC_single-wasp.predictions.h5").parent),
+        # empty string, simulate user canceling the dialog
+        "",
+    ],
+)
+def test_on_browse_clicked(file_path, make_napari_viewer_proxy, mocker):
+    """Test that the _on_browse_clicked method correctly sets the
+    file path in the QLineEdit widget (file_path_edit).
+    The file path is provided by mocking the return of the
+    QFileDialog.getOpenFileName method.
+    """
+    # Instantiate the napari viewer and the poses loader widget
+    viewer = make_napari_viewer_proxy()
+    poses_loader_widget = PosesLoader(viewer)
+
+    # Mock the QFileDialog.getOpenFileName method to return the file path
+    mocker.patch(
+        "movement.napari._loader_widgets.QFileDialog.getOpenFileName",
+        return_value=(file_path, None),  # tuple(file_path, filter)
```

Comment:
Could we test the filters as well, e.g. that we expect for "DeepLabCut": ["*.h5", "*.csv"] ?

Reply:
I added a separate test for these, in [6483915](https://github.com/neuroinformatics-unit/movement/pull/253/commits/6483915489d602f91f1bfce1cb60d3d300119636).